test/integration/buyers/applications_controller_test.rb

```diff
@@ -45,5 +45,17 @@ def setup
       page = Nokogiri::HTML::Document.parse(response.body)
       refute page.xpath("//tr").text.match /Service/
     end
+
+    test 'user do not have permission to access the service' do
+      plan = @provider.services.first.application_plans.create(name: 'plan')
+      buyer = FactoryBot.create(:buyer_account, provider_account: @provider)
+
+      User.any_instance.stubs(accessible_services: [])
+
+      # post admin_api_account_applications_path(account_id: @buyer.id, plan_id: @plan.id, format: :xml)
+      post admin_buyers_applications_path(account_id: buyer.id, plan_id: plan.id, format: :xml)
```

Comment:
Remove that format because it is for the UI and not the API so it won't be XML

Reply:
Moving the test to this file leads to very weird errors 😕 
```
# User.any_instance.stubs(accessible_services: [])

NoMethodError: undefined method `where_values_hash' for []:Array
```

```
# post admin_buyers_applications_path(account_id: buyer.id, plan_id: plan.id)

ActionController::UrlGenerationError: No route matches {:action=>"new", :controller=>"buyers/applications"} missing required keys: [:account_id]
from /Users/jgallaso/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/actionpack-5.0.7.2/lib/action_dispatch/journey/formatter.rb:51:in `generate'
```